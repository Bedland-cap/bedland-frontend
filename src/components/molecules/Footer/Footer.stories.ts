import type { Meta, StoryObj } from '@storybook/react';
import Footer from './Footer';

// docs: https://storybook.js.org/docs/7.0/react/writing-stories/introduction
const meta = {
  title: 'molecules/Footer',
  component: Footer,
  tags: ['autodocs'],
} satisfies Meta<typeof Footer>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = { args: { backgroundColor: '#3F4476' } };
