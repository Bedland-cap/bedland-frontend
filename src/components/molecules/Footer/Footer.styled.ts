import styled from 'styled-components';

const FooterBox = styled.div<{ backgroundColor: string }>`
  align-items: center;
  background-color: ${(props) => props.backgroundColor};
  color: #ffffff;
  display: flex;
  height: 3.3125rem;
  justify-content: center;
  width: 100vw;
`;

export default FooterBox;
