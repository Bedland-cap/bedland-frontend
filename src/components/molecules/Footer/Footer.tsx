import FooterBox from './Footer.styled';

const Footer = ({ backgroundColor }: { backgroundColor: string }) => (
  <FooterBox backgroundColor={backgroundColor}>Â©BedLand 2023</FooterBox>
);

export default Footer;
