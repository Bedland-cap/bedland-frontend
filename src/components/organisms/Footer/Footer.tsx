import Typography from 'components/atoms/Typography/Typography';
import { useContext } from 'react';
import styled from 'styled-components';
import { Palette } from 'theme/theme.types';
import { ThemeContext } from 'theme/ThemeContext';
import { FOOTER_HEIGHT } from 'utils/constans';

const FooterBox = styled.div<Palette>`
  align-items: center;
  background-color: ${({ palette }) => palette.footer};
  display: flex;
  height: ${FOOTER_HEIGHT}px;
  justify-content: center;
`;

const Footer = () => {
  const { palette } = useContext(ThemeContext);
  return (
    <FooterBox palette={palette}>
      <Typography variant='caption' color='light' style={{ fontWeight: 600 }}>
        Â©BedLand 2023
      </Typography>
    </FooterBox>
  );
};

export default Footer;
