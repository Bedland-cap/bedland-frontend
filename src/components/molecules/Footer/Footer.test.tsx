import renderer from 'react-test-renderer';

import Footer from './Footer';

describe('Footer', () => {
  it('renders correctly', () => {
    const snapshot = renderer
      .create(<Footer backgroundColor='#ffffff' />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
