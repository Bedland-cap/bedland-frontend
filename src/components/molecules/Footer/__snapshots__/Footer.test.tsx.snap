// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div
  className="sc-bcXHqe jQetJk"
>
  Â©BedLand 2023
</div>
`;
